feat: set and update environment variables

picccard opened the pull request:
This pull request adds logic for setting and updating environment variables in the deployment script. The main change is the introduction of a new section that ensures environment variables are correctly set for each environment, skipping updates when values have not changed. The process will never remove any variables, this must be done manually.

**Environment variable management:**

* Added a new section to `src/Deploy-GitHubRepository.ps1` that iterates over `environment.variables` and sets each variable for the specified environment using the GitHub CLI, only updating variables if their values have changed.

Example metadata.json:
```
{
  "$schema": "https://raw.githubusercontent.com/climpr/climpr-schemas/refs/heads/feat/set-gh-env-var-in-metadata-json/schemas/v1.0.0/lz-management/metadata.json#",
  "organization": "example-org",
  "repoName": "example-repo",
  "environments": [
    {
      "name": "test",
      "variables": [
```

<strike>

```json
        { "name": "TEST_1", "value": "value1" },
        { "name": "TEST_2", "value": "value2" }
```

</strike>

```
        "VAR_1": "value_1",
        "VAR_2": "value_2"
      ]
    },
    {
      "name": "dev",
      "variables": [
        { "DEV_1": "value1" },
        { "DEV_2": "value2_2" }
      ]
    }
  ]
}
```

## preserve the "last updated" timestamp
preserve the "last updated" timestamp for non-changed varaibles by only setting variable value if the provided value differs from the existing value:
<img width="813" height="258" alt="image" src="https://github.com/user-attachments/assets/9972de11-2216-4cf0-84ea-9bc45c0788d8" />

## Example run output
```
...
Set environment branch policy patterns: dev
- Skipping. Branch policy patterns can only be applied when the environment protection rule deployment_branch_policy.custom_branch_policies is set to 'true'.
Create environment variables: dev
- [dev] Skipping environment variable, value has not changed [name=TEST_444, value=value112].
- [dev] Updated environment variable [name=TEST_55, value=value2251].
- [dev] Created environment variable [name=TEST_6, value=value1].
```

## metadata.json schema
- climpr/climpr-schemas#8

picccard committed:
feat: add functionality to create environment variables in GitHub repository

picccard committed:
only update or set value if the value differs from the existing value

picccard commented on src/Deploy-GitHubRepository.ps1:
```diff
@@ -785,6 +785,32 @@ if (!$lzConfig.decommissioned) {
+        #region
+        Write-Host "Create environment variables: $($environmentName)"
+
+        foreach ($environmentVariable in $environment.variables) {
+            $name = $environmentVariable.name
+            $newValue = $environmentVariable.value
+
+            $currentValue = gh variable get $name --repo $org/$repo --env $environmentName
+
+            if ($currentValue -eq $newValue) {
+                Write-Host "Skip setting repository environment [$environmentName] variable [$name] value. No change."
+                continue
+            }
+
+            gh variable set $name `
+                --repo $org/$repo `
+                --body $newValue `
+                --env $environmentName
+            Write-Host "Set repository environment [$environmentName] variable [$name] value [$newValue]"
```
it will not cause an error... if no variables is provided the value will be null and the loop will be skipped.

audhage reviewed (commented):
Love the addition. However, we try to stick with `Invoke-GitHubCliApiMethod` across the board to ensure consistent error handling and calling behavior. Is there a specific reason this is not used here?

picccard commented:
I'll change to make use of it. This was initially just a raw copy from the deploy-lz script.

https://github.com/climpr/deploy-landing-zone/blob/main/src/Deploy-AzureLandingZone.ps1#L166

picccard committed:
make use of cmdlet `Invoke-GitHubCliApiMethod`

picccard commented:
- Updated PR summary with example run
- Updated PR summary json example to have the variables be a json object instead of list
- Changed code to make use of `Invoke-GitHubCliApiMethod`
